beacon-chain/blockchain/process_block.go

```diff
@@ -213,11 +210,27 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 		}).Debug("Head block is not the received block")
 	}
 	newBlockHeadElapsedTime.Observe(float64(time.Since(start).Milliseconds()))
+	// Checks if the current blockRoot ( which is also the head root) is new.
+	// This check must come before forkchoiceUpdateWithExecution because it saves the new head.
+	isNewHead := s.isNewHead(headRoot)
 
+	// updates FCU and prunes the attestations from conical block.
 	if err := s.forkchoiceUpdateWithExecution(ctx, headRoot, s.CurrentSlot()+1); err != nil {
 		return err
 	}
 
+	if isNewHead {
```

Comment:
Perhaps we can consider moving this block to `ReceiveBlock` to alleviate the cognitive value of `onBlock`

Reply:
may need bigger refactors to do this, blockRoot passed in is checked against headRoot first as a comparison that's why this check if afterwards. would need to move more things around to enable this. maybe we can keep it here until I move bigger pieces.

Reply:
moved it based on slack discussion.